internal/controller/storageclaim_controller.go

```diff
@@ -594,3 +648,17 @@ func (r *StorageClaimReconciler) hasVolumeSnapshotContents() (bool, error) {
 
 	return false, nil
 }
+
+func (r *StorageClaimReconciler) hasVolumeGroupSnapshotContents() (bool, error) {
+	vscList := &groupsnapapi.VolumeGroupSnapshotContentList{}
+	if err := r.list(vscList, client.MatchingFields{vscClusterIDIndexName: r.storageClaimHash}); err != nil {
```

Comment:
Pls add IndexField on this object as well while setting manager similar to VOlumeSnapshotContent.